themes/10up-theme/src/Assets.php

```diff
@@ -0,0 +1,238 @@
+<?php
+/**
+ * Assets module.
+ *
+ * @package TenUpTheme
+ */
+
+namespace TenUpTheme;
+
+use TenupFramework\Module;
+use TenupFramework\ModuleInterface;
+use TenUpTheme\Traits\GetAssetInfo;
+
+/**
+ * Assets module.
+ *
+ * @package TenUpTheme
+ */
+class Assets implements ModuleInterface {
+
+	use Module;
+	use GetAssetInfo;
+
+	/**
+	 * Can this module be registered?
+	 *
+	 * @return bool
+	 */
+	public function can_register() {
+		return true;
+	}
+
+	/**
+	 * Register any hooks and filters.
+	 *
+	 * @return void
+	 */
+	public function register() {
+		add_action( 'wp_enqueue_scripts', [ $this, 'scripts' ] );
+		add_action( 'admin_enqueue_scripts', [ $this, 'admin_styles' ] );
+		add_action( 'admin_enqueue_scripts', [ $this, 'admin_scripts' ] );
+		add_action( 'enqueue_block_editor_assets', [ $this, 'enqueue_block_editor_scripts' ] );
+		add_action( 'wp_enqueue_scripts', [ $this, 'styles' ] );
+		add_action( 'wp_head', [ $this, 'js_detection' ], 0 );
+		add_action( 'wp_head', [ $this, 'embed_ct_css' ], 0 );
+
+		add_filter( 'script_loader_tag', [ $this, 'script_loader_tag' ], 10, 2 );
+	}
+
+
+	/**
+	 * Enqueue scripts for front-end.
+	 *
+	 * @return void
+	 */
+	public function scripts() {
+
+		/**
+		 * Enqueuing frontend.js is required to get css hot reloading working in the frontend
+		 * If you're not shipping any front-end js wrap this enqueue in a SCRIPT_DEBUG check.
+		 */
+		wp_enqueue_script(
+			'frontend',
+			TENUP_THEME_TEMPLATE_URL . '/dist/js/frontend.js',
+			$this->get_asset_info( 'frontend', 'dependencies' ),
+			$this->get_asset_info( 'frontend', 'version' ),
+			true
+		);
+
+		/**
+		 * Enqueuing shared.js is required to get css hot reloading working in the frontend
+		 * If you're not shipping any shared js wrap this enqueue in a SCRIPT_DEBUG check.
+		 */
+
+		/*
+		 * Uncoment this to use the shared.js file.
+			wp_enqueue_script(
+				'shared',
+				TENUP_THEME_TEMPLATE_URL . '/dist/js/shared.js',
+				$this->get_asset_info( 'shared', 'dependencies' ),
+				$this->get_asset_info( 'shared', 'version' ),
+				true
+			);
+		*/
```

Comment:
Doesn't need to be in this PR but I'm in favor of removing all the "shared" stuff here also...